src/project-config/resolve.ts

```diff
@@ -9,7 +9,7 @@ function resolve(
   overrides: PartialDeep<IProjectConfig> = {},
   projectDir: string = ''
 ): IResolvedProjectConfig {
-  const resolvePath = (p: string) => path.resolve(projectDir || '', p);
```

Comment:
It seems to me that there should be a `null` to use a default value, otherwise, it will use `cwd` as directory.
Are you sure you test it with all cases?

Reply:
I think I'll leave the fix of `resolvePath()` to only scripts/include so we don't have to think in the other cases. Later we can figured out if the fix is safe to use on the other functions.

Using and empty string or null will work in any of the cases, because the guard clause uses `!` which will give false on `'', undefined, false, 0, or NaN`. Not sure why you say it will use `cwd` directory, if the path given is empty, it will not resolve to anything.

The tests I did was with a empty string, the default `./dist` value and and absolute path like `C:/Users/rtxa/Documents/Servidores HL/AG Mod X/valve/addons/amxmodx/plugins` and they all work fine.

Reply:
`resolvePath` is used for all configuration paths, for example in case it will be equal to `''` It can me it can lead to unsafe behavior in this case:
https://github.com/Hedgefog/node-amxxpack/blob/master/src/cli/services/project-creator.ts#L183

I don't remember all details, but I remember that I refused to make a quick fix because of it.

Reply:
`path.resolve('', 'foo/bar')` will return `/cwd/foo/bar` as the result
`path.resolve(null, 'foo/bar')` will throw an exception
`path.resolve('', '/foo/bar')` will return `/foo/bar`

Somewhere I wrote the logic in a way, that null values will replace with default values, and empty strings will use as the param value. I don't remember if I wrote it for the fix or if I had this logic before, but It probably should be there.

Reply:
Okay, don't worry, in the mean time I'll use my fork, hope you are better and better times will come for sure.
 
For me, this issue with `resolvePath` could be a problem of consistency and the ambiguities by using null values checks in some places, or checking for undefined or an empty string, etc. I suggest you to read some articles about null safety, even VSCode team developers using TypeScript have issues with it and try to avoid them.

https://www.split.io/blog/bias-against-null-javascript/
https://code.visualstudio.com/blogs/2019/05/23/strict-null
https://blog.jetbrains.com/idea/2017/08/code-smells-null/